### PR TITLE
Add custom service config

### PR DESCRIPTION
**Pull request by sameeragunarathne**

### CODE REVIEW

File: transformation/v2-to-fhirr4-service/README.md [hunk 1]
```diff
@@ -29,6 +29,39 @@ You do not have to write code from scratch but reuse these existing services whe
     --data-raw 'MSH|^~\&|EPIC|EPICADT|SMS|SMSADT|202211031408|CHARRIS|ADT^A01|1817457|D|2.8'
     ```
 
+## Customization
+
+You can customize the v2 segments to FHIR transformation logic by implement a service with custom mappings and configure the v2-to-fhirr4-service to invoke the custom service. This custom service can be implemented from Ballerina by using the template or use the OAS definition provided(resources/custom-mapping-service-openapi.yaml) to implement your service using any other technology. Following is the command to initiate the Ballerina template.
```
Comment (nirmal070125) [inline transformation/v2-to-fhirr4-service/README.md:34]: by implementing

File: transformation/v2-to-fhirr4-service/README.md [hunk 2]
```diff
@@ -29,6 +29,39 @@ You do not have to write code from scratch but reuse these existing services whe
...
+```toml
+[serviceConfig]
+baseUrl = "http://localhost:9091/v2tofhir"
+
+[serviceConfig.segmentToAPI.NK1]
+path = "/segment/nk1"
+
+[serviceConfig.segmentToAPI.PID]
+path = "/segment/pid"
+useDefault = true
```
Comment (nirmal070125) [inline transformation/v2-to-fhirr4-service/README.md:51]: can't we use just key-value (segment-path) pairs? If it is not present, we use the default mapping.

**Review by nirmal070125 — CHANGES_REQUESTED** [review]

File: transformation/v2-to-fhirr4-service/README.md [hunk 3]
```diff
@@ -29,6 +29,34 @@ You do not have to write code from scratch but reuse these existing services whe
...
+You can configure the v2-to-fhir-service to point your custom service to invoke the mapping endpoints. Following is a sample configuration structure to invoke a custom service.
+
+```toml
+[serviceConfig]
+baseUrl = "http://localhost:9091/v2tofhir"
+
+[serviceConfig.segmentToAPI]
+NK1 = "/segment/nk1"
+PID = "/segment/pid"
+```
```
Comment (nirmal070125) [inline transformation/v2-to-fhirr4-service/README.md:49]: What's the file name of this config? Is it Config.toml? Let's mention it.

File: transformation/v2-to-fhirr4-service/README.md [hunk 4]
```diff
@@ -29,6 +29,34 @@ You do not have to write code from scratch but reuse these existing services whe
...
+
+
+| Table | Field | Description |
+|-------|-------|-------------|
+| `serviceConfig`| `baseUrl` | The base URL of the custom service. |
+| `serviceConfig.segmentToAPI`| `<segmentName>` | The path of the custom service endpoint against the each customized segment name. |
+
+> **Note:** If you deploy this to [Choreo](https://wso2.com/choreo/), you can apply configurations through the [config editor](https://wso2.com/choreo/docs/devops-and-ci-cd/manage-configurations-and-secrets/#manage-ballerina-configurables) in the Choreo console.
```
Comment (nirmal070125) [inline transformation/v2-to-fhirr4-service/README.md:57]: If you deploy this to [Choreo](https://wso2.com/choreo/) (refer to the ## [Optional] Deploy in Choreo section), ...

File: transformation/v2-to-fhirr4-service/README.md (hunk 3 above)
Comment (sameeragunarathne) [reply transformation/v2-to-fhirr4-service/README.md:49]: yes. fixed

**Review by nirmal070125 — DISMISSED** [review]

**Review by nirmal070125 — APPROVED** [review]

**Pull request merged**